### ISSUE TITLE
undefined reference to `dlopen' - libcrypto.a

### ISSUE THREAD

**Issue by dsteckbeck** [FreeTDS/freetds#54]
When trying to compile 0.95.87 against a static openssl, I'm getting an error when freetds is compiling the fisql tool:

`Making all in fisql
make[3]: Entering directory `/home/dsteckbeck/apache-php/freetds-0.95.87/src/apps/fisql'
  CC       fisql.o
  CC       edit.o
  CC       handlers.o
  CC       interrupt.o
  CCLD     fisql
/home/dsteckbeck/apache-php/libs/lib/libcrypto.a(dso_dlfcn.o): In function `dlfcn_globallookup':
dso_dlfcn.c:(.text+0x31): undefined reference to `dlopen'
dso_dlfcn.c:(.text+0x44): undefined reference to `dlsym'
dso_dlfcn.c:(.text+0x4f): undefined reference to `dlclose'
/home/dsteckbeck/apache-php/libs/lib/libcrypto.a(dso_dlfcn.o): In function `dlfcn_pathbyaddr':
dso_dlfcn.c:(.text+0xa0): undefined reference to `dladdr'
dso_dlfcn.c:(.text+0x101): undefined reference to `dlerror'
/home/dsteckbeck/apache-php/libs/lib/libcrypto.a(dso_dlfcn.o): In function `dlfcn_bind_func':
dso_dlfcn.c:(.text+0x454): undefined reference to `dlsym'
dso_dlfcn.c:(.text+0x512): undefined reference to `dlerror'
/home/dsteckbeck/apache-php/libs/lib/libcrypto.a(dso_dlfcn.o): In function `dlfcn_bind_var':
dso_dlfcn.c:(.text+0x584): undefined reference to `dlsym'
dso_dlfcn.c:(.text+0x642): undefined reference to `dlerror'
/home/dsteckbeck/apache-php/libs/lib/libcrypto.a(dso_dlfcn.o): In function `dlfcn_unload':
dso_dlfcn.c:(.text+0x69f): undefined reference to `dlclose'
/home/dsteckbeck/apache-php/libs/lib/libcrypto.a(dso_dlfcn.o): In function `dlfcn_load':
dso_dlfcn.c:(.text+0x754): undefined reference to `dlopen'
dso_dlfcn.c:(.text+0x7c2): undefined reference to `dlclose'
dso_dlfcn.c:(.text+0x7f2): undefined reference to `dlerror'
collect2: ld returned 1 exit status
make[3]: *** [fisql] Error 1
make[3]: Leaving directory `/home/dsteckbeck/apache-php/freetds-0.95.87/src/apps/fisql'
`

The configure options are:
`
./configure --prefix=/home/dsteckbeck/apache-php/libs --with-pic --with-tdsver=7.1 --enable-static --enable-msdblib --disable-shared --disable-odbc --disable-pool --disable-server --with-openssl=/home/dsteckbeck/apache-php/libs
`

This is with OpenSSL 1.0.2f. Previous builds of FreeTDS 0.92.79 and OpenSSL 1.0.1j compiled together just fine.

**Comment by dsteckbeck** [comment]
So, the link line was missing '-ldl'. I saw in the Makefile and configure scripts, it was supposed to work out the dlopen link switch, and apparently the new code is requiring it (the Makefile in the older version I referenced looked similar). As a workaround, I exported LDFLAGS to set it manually for the build.
```
export LDFLAGS="-L$LIBDIR/lib -ldl"
```
If dlopen is now a requirement for statically linked builds, the configure script should work this out properly and generate correct Makefile for the tools requiring it.
